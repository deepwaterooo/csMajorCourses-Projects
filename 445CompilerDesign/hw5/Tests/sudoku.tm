* C- compiler version C-F07
* Author: Heyan Huang
* Backend adapted from work by Jorge Williams (2001)
* File compiled:  sudoku.tm
* Nov 29, 2007
* BEGIN function input
  1:     ST  3,-1(1) 	Store return address 
  2:     IN  2,2,2 	Grab int input 
  3:     LD  3,-1(1) 	Load return address 
  4:     LD  1,0(1) 	Adjust fp 
  5:    LDA  7,0(3) 	Return 
* END of function input
* BEGIN function output
  6:     ST  3,-1(1) 	Store return address 
  7:     LD  3,-2(1) 	Load parameter 
  8:    OUT  3,3,3 	Output integer 
  9:    LDC  2,0(6) 	Set return to 0 
 10:     LD  3,-1(1) 	Load return address 
 11:     LD  1,0(1) 	Adjust fp 
 12:    LDA  7,0(3) 	Return 
* END of function output
* BEGIN function inputb
 13:     ST  3,-1(1) 	Store return address 
 14:    INB  2,2,2 	Grab bool input 
 15:     LD  3,-1(1) 	Load return address 
 16:     LD  1,0(1) 	Adjust fp 
 17:    LDA  7,0(3) 	Return 
* END of function inputb
* BEGIN function outputb
 18:     ST  3,-1(1) 	Store return address 
 19:     LD  3,-2(1) 	Load parameter 
 20:   OUTB  3,3,3 	Output bool 
 21:    LDC  2,0(6) 	Set return to 0 
 22:     LD  3,-1(1) 	Load return address 
 23:     LD  1,0(1) 	Adjust fp 
 24:    LDA  7,0(3) 	Return 
* END of function outputb
* BEGIN function outnl
 25:     ST  3,-1(1) 	Store return address 
 26:  OUTNL  3,3,3 	Output a newline 
 27:     LD  3,-1(1) 	Load return address 
 28:     LD  1,0(1) 	Adjust fp 
 29:    LDA  7,0(3) 	Return 
* END of function outnl
  0:    LDA  7,29(7) 	Jump to init 
* BEGIN Init
 30:     LD  0,0(0) 	Set the global pointer 
* BEGIN init of global array sizes
 31:    LDC  3,0(6) 	load size of array board
 32:     ST  3,-1(0) 	save size of array board
* END init of global array sizes
* BEGIN init of global array sizes
 33:    LDC  3,0(6) 	load size of array locked
 34:     ST  3,-2(0) 	save size of array locked
* END init of global array sizes
 35:    LDA  1,-2(0) 	set first frame at end of globals 
 36:     ST  1,0(1) 	store old fp (point to self) 
 37:    LDA  3,1(7) 	Return address in ac 
 38:    LDA  7,-39(7) 	Jump to main 
 39:   HALT  0,0,0 	DONE! 
* END Init
