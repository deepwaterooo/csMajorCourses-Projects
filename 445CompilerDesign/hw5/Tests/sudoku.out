ERROR(226): Symbol 'initBoard' is not defined.
ERROR(227): Symbol 'getInput' is not defined.
ERROR(228): Symbol 'solve' is not defined.
ERROR(229): Symbol 'showBoard' is not defined.
ERROR(228): Expecting Boolean test condition in if statement but got type (null).
Number of warnings: 0
Number of errors: 5
------------------------------------
